src/main/java/org/openlmis/stockmanagement/service/StockCardSummariesService.java

```diff
@@ -129,13 +130,13 @@ public StockCardSummaries findStockCards(StockCardSummariesV2SearchParams params
     permissionService.canViewStockCard(params.getProgramId(), params.getFacilityId());
 
     profiler.start("GET_APPROVED_PRODUCTS");
-    Page<OrderableDto> approvedProducts = approvedProductReferenceDataService
+    WrappedOrderablesDto approvedProducts = approvedProductReferenceDataService
         .getApprovedProducts(params.getFacilityId(), params.getProgramId(),
             params.getOrderableIds());
 
     profiler.start("FIND_ORDERABLE_FULFILL_BY_ID");
     Map<UUID, OrderableFulfillDto> orderableFulfillMap = orderableFulfillService.findByIds(
-        approvedProducts.getContent().stream().map(OrderableDto::getId).collect(toList()));
+        approvedProducts.getIdentifiersPage().getContent());
```

Comment:
why we return a page if in the end we only care about the content?

Reply:
Nice catch. I'll use pageable only for the orderables.